contracts/BalanceTrackerFixedPrice.sol

```diff
@@ -0,0 +1,332 @@
+        if (token == address (0)) {
+            // Wrap native tokens
+            _wrap(localCollectedFees);
+            // Transfer to Buy back burner
+            IToken(wrappedNativeToken).transfer(buyBackBurner, localCollectedFees);
+        } else {
+            IToken(token).transfer(buyBackBurner, localCollectedFees);
+        }
+
+        emit Drained(token, localCollectedFees);
+
+        _locked = 1;
+    }
+
+    function _withdraw(address token, uint256 balance) internal {
+        bool success;
+        // Transfer mech balance
+        if (token == address(0)) {
+            // solhint-disable-next-line avoid-low-level-calls
+            (success, ) = msg.sender.call{value: balance}("");
+        } else {
+            IToken(token).transfer(msg.sender, balance);
+        }
+
+        // Check transfer
+        if (!success) {
+            revert TransferFailed(token, address(this), msg.sender, balance);
+        }
+    }
+
+    /// @dev Processes mech payment by withdrawing funds.
+    function processPayment(address token) external returns (uint256 mechPayment, uint256 marketplaceFee) {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        // Get mech balance
+        uint256 balance = mapMechBalances[msg.sender][token];
+        // TODO limits?
+        if (balance == 0) {
+            revert ZeroValue();
+        }
+
+        // Calculate mech payment and marketplace fee
+        uint256 fee = IMechMarketplace(mechMarketplace).fee();
+        marketplaceFee = (balance * fee) / 10_000;
+        mechPayment = balance - marketplaceFee;
+
+        // Check for zero value, although this must never happen
+        if (mechPayment == 0) {
+            revert ZeroValue();
+        }
+
+        // Adjust marketplace fee
+        mapCollectedFees[token] += marketplaceFee;
+
+        // Clear balances
+        mapMechBalances[msg.sender][token] = 0;
+
+        // Process withdraw
+        _withdraw(token, balance);
+
+        emit Withdraw(msg.sender, token, balance);
+
+        _locked = 1;
+    }
+
+    /// @dev Withdraws funds for a specific requester account.
+    function withdraw(address token) external {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        // Get account balance
+        uint256 balance = mapRequesterBalances[msg.sender][token];
+        // TODO limits?
+        if (balance == 0) {
+            revert ZeroValue();
+        }
+
+        // Clear balances
+        mapRequesterBalances[msg.sender][token] = 0;
+
+        // Process withdraw
+        _withdraw(token, balance);
+
+        emit Withdraw(msg.sender, token, balance);
+
+        _locked = 1;
+    }
+
+    // Deposits token funds for requester.
+    function deposit(address token, uint256 amount) external {
+        // TODO Accept deposits from mechs as well?
```

Comment:
As mentioned above, whilst this is a nice additional feature, the default of one-off payments also needs to be retained for fixed price!